test/frontend_test.dart

```diff
@@ -0,0 +1,457 @@
+          await workoutVarietyFilterDropdown.click();
+          var jogaFilterOption = await driver.findElement(const By.xpath(
+              '*//div[contains(@class, "options")]/div/material-select-'
+                  'dropdown-item/span[contains(text(), "jóga")]'));
+          await jogaFilterOption.click();
+          await waitForAngular(driver);
+          await sleep2();
+          var filteredVideoItemsLength = await driver
+              .findElements(const By.cssSelector(
+                  'video-summary-list > div.training-type-list > div '
+                  '> video-summary > div.video-summary'))
+              .length;
+          expect(await isXAtLeastValueOfY(filteredVideoItemsLength, 2), true);
+        });
+
+        test('time filter options work', () async {
+          var timeFilterDropdown = await driver.findElement(
+              const By.xpath('*//tag-set-filter/material-dropdown-select/'
+                  'dropdown-button/div/span[contains(text(), "Hossz")]'));
+          await timeFilterDropdown.click();
+          var thirtyMinFilterOption = await driver.findElement(const By.xpath(
+              '*//div[contains(@class, "options")]/div/material-select-'
+                  'dropdown-item/span[contains(text(), "45 perc")]'));
+          await thirtyMinFilterOption.click();
+          await waitForAngular(driver);
+          await sleep2();
+          var filteredVideoItemsLength = await driver
+              .findElements(const By.cssSelector(
+                  'video-summary-list > div.training-type-list > div '
+                  '> video-summary > div.video-summary'))
+              .length;
+          expect(await isXAtLeastValueOfY(filteredVideoItemsLength, 6), true);
+        });
+
+        test('text filter options work', () async {
+          var textFilter = await driver.findElement(
+              const By.xpath("*//input[contains(@class, 'search-input')]"));
+          await textFilter.click();
+          await textFilter.sendKeys("FORMATORNA");
+          await waitForAngular(driver);
+          await sleep2();
+          var filteredVideoItemsLength = await driver
+              .findElements(const By.cssSelector(
+                  'video-summary-list > div.training-type-list > div '
+                  '> video-summary > div.video-summary'))
+              .length;
+          expect(await isXAtLeastValueOfY(filteredVideoItemsLength, 2), true);
+        });
+      });
+    });
+
+    group("product page megvasarlom tests |", () {
+      setUpAll(() async {
+        await loginUserWithCredentials(
+            newUserEmail, allUserPassword, driver, true);
+        await waitForAngular(driver);
+        await navigateToPage(driver, 'profile/product/probaidoszak');
+        await waitForAngular(driver);
+        await sleep5();
+      });
+
+      test('clicking on megvasarlom throws no errors | the gwl', () async {
+        bool hasErrors = false;
+        await clickMegvasarlom(
```

Comment:
Do not mix languages in variable, method and class names. I'd suggest `clickPurchase`.